File: wk4/java/forcedchoice.java
// problem link: https://open.kattis.com/problems/forcedchoice

import java.io.*;
import java.math.*;
import java.util.*;

public class forcedchoice {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        int[] NPS = Arrays.stream(r.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        while(NPS[2]-- > 0) {
            int[] a = Arrays.stream(r.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            boolean flag = true;
            for(int i = 1; i < a.length && flag; i++)
                if(a[i] == NPS[1]) {
                    System.out.println("KEEP");
                    flag = false;
                }
            if(flag)
                System.out.println("REMOVE");
        }
    }
}

File: wk4/cpp/forcedchoice.cpp
// problem link: https://open.kattis.com/problems/forcedchoice

#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    
	int n, p, s;
	cin >> n >> p >> s;
	while(s--) {
	    int m, t;
	    cin >> m;
	    bool f = false;
	    while(m-- > 0) {
	        cin >> t;
	        if(t == p)
	            f = true;
	    }
	    cout << (f ? "KEEP\n" : "REMOVE\n");
	}
	    
	return 0;
}